### PR TITLE
tests: Add more GPU-AV BDA tests

### PR DESCRIPTION
**Pull request by spencer-lunarg**
Adds all the tests from `inst_buff_addr_check_test.cpp` in `spirv-opt` here to be tested as an end-to-end test for GPU-AV

### CODE REVIEW

**Review by arno-lunarg — APPROVED** [review]
LGTM, but the code for each test is essentially the same, it needs to be refacto

**Comment by spencer-lunarg** [comment]
> LGTM, but the code for each test is essentially the same, it needs to be refacto

yep, the goal was to just get "some" tests in, now that I have a pattern, and need to add more, will DRY it up

File: tests/unit/gpu_av_buffer_device_address_positive.cpp [hunk 1]
```diff
@@ -96,3 +90,277 @@ TEST_F(PositiveGpuAVBufferDeviceAddress, Basic) {
...
+    m_default_queue->wait();
+
+    in_buffer_ptr = (uint8_t *)in_buffer.memory().map();
+    float output = *((float *)(in_buffer_ptr + sizeof(VkDeviceAddress)));
+    in_buffer.memory().unmap();
+    ASSERT_TRUE(output == expected_output);
+}
+
+TEST_F(PositiveGpuAVBufferDeviceAddress, StructLoadPadded) {
+    TEST_DESCRIPTION("Do a OpLoad through a padded struct PhysicalStorageBuffer");
```
Comment (filnet) [inline tests/unit/gpu_av_buffer_device_address_positive.cpp:230]: Three tests have the same description.

Comment (spencer-lunarg) [reply tests/unit/gpu_av_buffer_device_address_positive.cpp:230]: good catch, will update

**Pull request merged**